src/contracts/exchange/NiftyswapExchange20.sol

```diff
@@ -0,0 +1,810 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.7.4;
+pragma experimental ABIEncoderV2;
+import "../interfaces/INiftyswapExchange20.sol";
+import "../utils/ReentrancyGuard.sol";
+import "@0xsequence/erc-1155/contracts/interfaces/IERC20.sol";
+import "@0xsequence/erc-1155/contracts/interfaces/IERC165.sol";
+import "@0xsequence/erc-1155/contracts/interfaces/IERC1155.sol";
+import "@0xsequence/erc-1155/contracts/interfaces/IERC1155TokenReceiver.sol";
+import "@0xsequence/erc-1155/contracts/tokens/ERC1155/ERC1155MintBurn.sol";
+import "@uniswap/lib/contracts/libraries/TransferHelper.sol";
+
+/**
+ * This Uniswap-like implementation supports ERC-1155 standard tokens
+ * with an ERC-20 based token used as a currency instead of Ether.
+ *
+ * Liquidity tokens are also ERC-1155 tokens you can find the ERC-1155
+ * implementation used here:
+ *    https://github.com/horizon-games/multi-token-standard/tree/master/contracts/tokens/ERC1155
+ *
+ * @dev Like Uniswap, tokens with 0 decimals and low supply are susceptible to significant rounding
+ *      errors when it comes to removing liquidity, possibly preventing them to be withdrawn without
+ *      some collaboration between liquidity providers.
+ */
+contract NiftyswapExchange20 is ReentrancyGuard, ERC1155MintBurn, INiftyswapExchange20 {
+  using SafeMath for uint256;
+
+  /***********************************|
+  |       Variables & Constants       |
+  |__________________________________*/
+
+  // Variables
+  IERC1155 internal token;                        // address of the ERC-1155 token contract
+  address internal currency;                      // address of the ERC-20 currency used for exchange
+  address internal factory;                       // address for the factory that created this contract
```

Comment:
```
  IERC1155 internal token;                        // address of the ERC-1155 token contract
  address internal currency;                      // address of the ERC-20 currency used for exchange
  address internal factory;      
```

Are there any reasons for not defining these variables as `immutable`?

Reply:
Only because they weren't in `NiftwapExchange.sol`. Will update